Orange/data/util.py

```diff
@@ -18,11 +18,15 @@ def one_hot(values, dtype=float):
     result
         2d array with ones in respective indicator columns.
     """
+    if not len(values):
+       return np.zeros((0, 0), dtype=dtype)
     return np.eye(np.max(values) + 1, dtype=dtype)[np.asanyarray(values, dtype=int)]
 
 
 def scale(values, min=0, max=1):
     """Return values scaled to [min, max]"""
+    if not len(values):
+        return np.array([], dtype=values.dtype)
```

Comment:
`dtype=float`is the case even when `scale(np.arange(10, dtype=int))`. Also won't fail on values an empty list.